cpp/src/neighbors/mg/snmg.cuh

```diff
@@ -119,6 +143,9 @@ void build(const raft::resources& clique,
       cuvs::neighbors::build(dev_res, ann_if, index_params, partition);
       resource::sync_stream(dev_res);
     }
+
+    // Restore original thread count
+    omp_set_num_threads(saved_omp_threads);
```

Comment:
I'm a bit skeptical of modifying variables like this based on env variables for the entire process. Rather we can specify threads for a particular loop such as `#pragma omp parallel for num_threads(n)` (this is for the main loop) in this PR.